### PR TITLE
Added Constellation method `to_inst`

### PR DESCRIPTION
**Pull request by aburrell**
# Description

Addresses #764 by adding the last of the targeted methods, `to_inst`, to the Constellation.  Supporting methods/functions were added to Instrument and utils.coords.  Partially addresses #796.

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

# How Has This Been Tested?

Added an example in the tutorial and a unit test.

**Test Configuration**:
* Operating system: OS X Big Sur
* Version number: Python 3.9
* Any details about your local setup that are relevant

# Checklist:

- [x] Make sure you are merging into the ``develop`` (not ``main``) branch
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] Add a note to ``CHANGELOG.md``, summarizing the changes

If this is a release PR, replace the first item of the above checklist with the release
checklist on the wiki: https://github.com/pysat/pysat/wiki/Checklist-for-Release

### CODE REVIEW

File: docs/tutorial/tutorial_constellation.rst [hunk 1]
```diff
@@ -47,3 +47,94 @@ The :py:attr:`~pysat._instrument.Instrument.name` values are: :py:data:`epam`,
...
+.. code:: python
+
+          import pysat
+          import pysatNASA
+
+	  # Initalize the ICON Constellation using the ICON module
+	  icon = pysat.Constellation(cont_module=pysatNASA.constellations.icon)
+
+	  # Display the results
+	  print(icon)
```
Comment (jklenzing) [inline docs/tutorial/tutorial_constellation.rst:69]:
```suggestion
    import pysat
    import pysatNASA

    # Initalize the ICON Constellation using the ICON module
    icon = pysat.Constellation(cont_module=pysatNASA.constellations.icon)

    # Display the results
    print(icon)
```
tabs are messed up here.  Displays correctly, copy and paste does not work

File: docs/tutorial/tutorial_constellation.rst [hunk 2]
```diff
@@ -47,3 +47,94 @@ The :py:attr:`~pysat._instrument.Instrument.name` values are: :py:data:`epam`,
...
+
+.. code:: python
+
+	  # Download today's data and load it into the Constellation
+	  ace_rt.download()
+	  ace_rt.load(date=ace_rt.today())
+
+	  # Convert the output to an Instrument
+	  rt_inst = ace_rt.to_inst()
+	  print(rt_inst)
```
Comment (jklenzing) [inline docs/tutorial/tutorial_constellation.rst:92]:
```suggestion
    # Download today's data and load it into the Constellation
    ace_rt.download()
    ace_rt.load(date=ace_rt.today())

    # Convert the output to an Instrument
    rt_inst = ace_rt.to_inst()
    print(rt_inst)
```

Comment (jklenzing) [reply docs/tutorial/tutorial_constellation.rst:92]: Getting secret character `^I` when I copy paste into terminal.

File: docs/tutorial/tutorial_constellation.rst [hunk 3]
```diff
@@ -47,3 +47,94 @@ The :py:attr:`~pysat._instrument.Instrument.name` values are: :py:data:`epam`,
...
+
+This yeilds:
```
Comment (jklenzing) [inline docs/tutorial/tutorial_constellation.rst:94]:
```suggestion
This yields:
```

**Review by jklenzing — CHANGES_REQUESTED** [review]

**Comment by jklenzing** [comment]
This is a very specific problem that probably doesn't need to be addressed, but I will note that the current invoking of the registry here requires all instruments in the registry to work.  This creates an issue if non-functional branches of other packages are open.

**Comment by aburrell** [comment]
> This is a very specific problem that probably doesn't need to be addressed, but I will note that the current invoking of the registry here requires all instruments in the registry to work.

Perhaps the best thing for us to do is put a note for developers that recommends they not update their registry until packages are merged in?

**Comment by jklenzing** [comment]
> > This is a very specific problem that probably doesn't need to be addressed, but I will note that the current invoking of the registry here requires all instruments in the registry to work.
> 
> Perhaps the best thing for us to do is put a note for developers that recommends they not update their registry until packages are merged in? Either way, it's outside the scope of this pull request.

I agree it's out of scope, just noting this because I found it strange.

**Comment by aburrell** [comment]
> I agree it's out of scope, just noting this because I found it strange.

I guess this reveals how frequently I use the registry 🦦

File: docs/tutorial/tutorial_constellation.rst [hunk 4]
```diff
@@ -47,3 +47,94 @@ The :py:attr:`~pysat._instrument.Instrument.name` values are: :py:data:`epam`,
...
+to create a Constellation of ICON EUV, FUV, IVM, and MIGHTI data (excluding the
+line-of-sight winds).
+
+.. code:: python
+
+    import pysat
+    import pysatNASA
+
+    # Initalize the ICON Constellation using the ICON module
+    icon = pysat.Constellation(cont_module=pysatNASA.constellations.icon)
```
Comment (jklenzing) [inline docs/tutorial/tutorial_constellation.rst:66]:
```suggestion
    icon = pysat.Constellation(const_module=pysatNASA.constellations.icon)
```

Comment (jklenzing) [reply docs/tutorial/tutorial_constellation.rst:66]: As currently defined at pysatNASA, the icon constellation will never load.  See https://github.com/pysat/pysatNASA/issues/152

**Review by jklenzing — COMMENTED** [review]
Tested with various instruments, looks like it works as expected. Some issues in the docs.

**Review by jklenzing — APPROVED** [review]
:otter:

**Pull request merged**